src/main/java/com/google/sps/servlets/GenerateServlet.java

```diff
@@ -26,16 +30,116 @@
 @WebServlet("/generate-hunt")
 public class GenerateServlet extends HttpServlet {
 
-  private static final String FILTER_ARRAY = "clicked-array";
+  private static final String PLACE_FILTERS = "user-places";
+  private static final String DIFF_FILTERS = "user-diff";
+
+  String[] allPlaces = {"Paris", "New York City", "San Francisco", "London", "Sydney", "Venice"};
+  String[] allDifficulties = {"Easy", "Medium", "Hard"};
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get array of clicked filters and convert to ArrayList<String>
-    Gson gson = new Gson();
-    HashSet<String> clickedFilters =
-        gson.fromJson(request.getParameter(FILTER_ARRAY), HashSet.class);
+
+    HashSet<String> userPlaces =
+        new Gson().fromJson(request.getParameter(PLACE_FILTERS), HashSet.class);
+    HashSet<String> userDifficultyStrings =
+        new Gson().fromJson(request.getParameter(DIFF_FILTERS), HashSet.class);
+
+    // Create fake destinations TODO: get destinations from datastore
+    ArrayList<Destination> allDestinations = createFakeDestinations();
+
+    // Convert difficulty level strings to Destination.Obscurity
+    HashSet<Destination.Obscurity> userDifficultyLevels = new HashSet();
+    for (String level : userDifficultyStrings) {
+      userDifficultyLevels.add(Destination.stringToEnum(level));
+    }
```

Comment:
You probably don't want to add UNDEFINED to your set, so you should probably check it before adding it to userDifficultyLevels.

Also you don't have to change it since what's below might be more difficult to read, but this can also be done with streams (my fav!): 
`HashSet <Destination.Obscurity> userDifficultyLevels = userDifficultyStrings.stream().map(Destination:: stringToEnum).filter(obscurity -> obscurity != Destination.Obscurity.UNDEFINED).collect(Collectors.toSet());`

What this is doing:
1.   Creating a stream so you can process each item
2.  Map the strings to the corresponding Obscurity level
3.  filtering out the UNDEFINED enums
4.  collecting it back into a set